### PR TITLE
clarify backward compatibility definitions; add SQL shell to list of interfaces

### LINKED ISSUE #14834: Feedback: API Support Policy
Raphael 'kena' Poss (knz) commented:

cc @taroface 
Page: https://cockroachlabs.com/docs/v22.1/api-support-policy.html

There's 3 things we need to enhance here.

### a detail I missed on the first review

> Backward-compatible changes
> The following list is not exhaustive:
> ...
> - Change in non-interactive cockroach sql shell input or output.

I'm not sure that's right?  In the *non-interactive* case, the API is stable, and so if we change the input or output in a way that breaks automation that would be a backward-incompatible change.
(But see point below for a clarification)

### need to add *definitions* for the terms "backward-incompatible change" and "backward-compatible change"

Currently the terms have *examples* but do not have *definitions*. We need to add them.

Suggestions:

- Backward-compatible change: when existing automation that relies *solely* on stable APIs continues to work as previously. If automation was relying on non-stable APIs (unstable or reserved), no guarantee is provided.

- Backward-incompatible change: when existing automation that relies *solely* on stable APIs needs to be updated to continue to work.

### Regarding the SQL shell

It's currently missing from the API table and should be there:

- the CLI flags of the SQL shell are a stable API, unless marked as unstable/preview
- the input client-side commands to the SQL shell are a stable API, unless marked as unstable/preview
- the output table formats of the SQL shell are a stable API, unless marked as unstable/preview
- any output of `cockroach sql` prefixed by `#` is an unstable API
- the behavior of `cockroach sql` when used *non-interactively*, given a set of CLI flags and output table format selection, insofar that only SQL input using stable features is executed and the `#`-prefixed output is ignored, is a stable API.
- the behavior of `cockroach sql` when used interactively is an unstable API.

In other words, it's safe to use `cockroach sql` in automation as long as the scripts only use a combination of stable features.

Jira Issue: DOC-5365

### PR DESCRIPTION
**Pull request by taroface**
Fixes #14834.

@knz Thanks for this feedback! I used simpler definitions for backward-in/compatibility according to my understanding of the policy as written. Our stability guarantees leave it possible for both backward-compatible and incompatible changes to be introduced to stable and unstable interfaces. But it's possible I am missing the point.

### CODE REVIEW

**Review by knz — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14843)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @taroface)

File: v22.1/api-support-policy.md [hunk 1]
```diff
@@ -62,26 +65,27 @@ A stable API may be assigned a new version number in two situations:
...
+| [Logging](logging-overview.html)                                        | Mixed    | Versioned concurrently with CockroachDB.                                               | Stability varies by [event type](eventlog.html). Structured events are stable and unstructured events are unstable.                                                                                                        | {{ site.data.products.core }}                                     |
+| `ccloud` CLI                                                            | Mixed    | Versioned independently from CockroachDB.                                              | Default output is unstable. Specify the `–json` argument in the CLI for stable output that follows the versioning scheme.                                                                                                  | {{ site.data.products.db }}                                       |
```
Comment (gemma-shay) [inline v22.1/api-support-policy.md:85]: Could link to /cockroachcloud/ccloud-get-started.html

File: v22.1/api-support-policy.md [hunk 2]
```diff
@@ -62,26 +65,27 @@ A stable API may be assigned a new version number in two situations:
...
+| [Health Endpoints](monitoring-and-alerting.html#health-endpoints)       | Stable   | No new versions forthcoming.                                                           |                                                                                                                                                                                                                            | All products                                                      |
+| [Prometheus Endpoint](monitoring-and-alerting.html#prometheus-endpoint) | Stable   | No new versions forthcoming.                                                           | Although this endpoint is not versioned, individual metrics may be added or removed in each CockroachDB release. No changes are expected to response format.                                                               | {{ site.data.products.dedicated }}, {{ site.data.products.core }} |
```
Comment (gemma-shay) [inline v22.1/api-support-policy.md:81]:
```suggestion
| [Prometheus endpoint](monitoring-and-alerting.html#prometheus-endpoint) | Stable   | No new versions forthcoming.                                                           | Although this endpoint is not versioned, individual metrics may be added or removed in each CockroachDB release. No changes are expected to response format.                                                               | {{ site.data.products.dedicated }}, {{ site.data.products.core }} |
```

File: v22.1/api-support-policy.md [hunk 3]
```diff
@@ -62,26 +65,27 @@ A stable API may be assigned a new version number in two situations:
...
+| [`cockroach sql` shell](cockroach-sql.html)                             | Mixed    | Versioned concurrently with CockroachDB.                                               | When used non-interactively, `cockroach sql` is stable unless your usage relies on unstable input or output. Any `cockroach sql` output prefixed by `#` is unstable. When used interactively, `cockroach sql` is unstable. | All products                                                      |
+| [Health Endpoints](monitoring-and-alerting.html#health-endpoints)       | Stable   | No new versions forthcoming.                                                           |                                                                                                                                                                                                                            | All products                                                      |
```
Comment (gemma-shay) [inline v22.1/api-support-policy.md:80]:
```suggestion
| [Health endpoints](monitoring-and-alerting.html#health-endpoints)       | Stable   | No new versions forthcoming.                                                           |                                                                                                                                                                                                                            | All products                                                      |
```

**Review by gemma-shay — APPROVED** [review]
Very minor nits, LGTM!

**Pull request merged**